lib/media.js

```diff
@@ -57,50 +52,106 @@ function findTop(str) {
  * @returns {string}
  */
 function splice(str, index, items) {
-  return str.substr(0, index) + _.flattenDeep(items).join('\n') + str.substr(index);
+  return str.substr(0, index) + items + str.substr(index);
+}
+
+/**
+ * Get the contents of a string that come after specified characters.
+ *
+ * @param  {string} str Any string to split
+ * @param  {string} dir The directory the file is in
+ * @return {string}
+ */
+function getFileName(str, dir) {
+  return str.split(dir)[1];
+}
+
+/**
+ * Retrieve the contents of a file
+ *
+ * @param  {array} fileArray   An array of file names
+ * @param  {string} targetDir  The directory to retrieve files from
+ * @param  {string} filePrefix The string that comes right before the file name
+ * @return {Promise}
+ */
+function getContentsOfFiles(fileArray, targetDir, filePrefix) {
+  var allPromises = [],
+    rootDir = process.cwd();
```

Comment:
I called it `rootDir` because it's running from the root of the `sites` repo which has access to all the files. Should it be something like `projectRoot`? Or `projectDir`? I defer to your naming because I'm the worst at naming.